drivers/ethernet/eth_stm32_hal.c

```diff
@@ -284,18 +319,8 @@ static int eth_tx(const struct device *dev, struct net_pkt *pkt)
 
 #if defined(CONFIG_ETH_STM32_HAL_API_V2)
 
-	for (;;) {
-		for (; ctx.tx_buffer_index < ETH_TXBUFNB; ctx.tx_buffer_index++) {
-			if (!dma_tx_buffer_used[ctx.tx_buffer_index]) {
-				dma_tx_buffer_used[ctx.tx_buffer_index] = true;
-				goto buffer_found;
-			}
-		}
-		ctx.tx_buffer_index = 0;
-		k_yield();
-	}
-
-buffer_found:
+	ctx.first_tx_buffer_index = allocate_tx_buffer();
+	buf_header = &dma_tx_buffer_header[ctx.first_tx_buffer_index];
```

Comment:
Can you squash this initial commit, so we don't review code that is later deleted ?